src/metaxy/metadata_store/postgres.py

```diff
@@ -0,0 +1,972 @@
+# ==============================================================================
+# Psycopg Compatibility Shim
+#
+# REASON: In some CI/Linux environments, psycopg's pure-Python fallback is
+# installed instead of the C-optimized version. The pure-Python version has a
+# different default behavior: it returns `bytes` for certain metadata query
+# results (like schema names and data types), whereas the C version correctly
+# decodes them to `str`. This inconsistency breaks upstream libraries like Ibis.
+#
+# SOLUTION: This code block acts as a compatibility shim. It monkey-patches the
+# `psycopg.Cursor` class to normalize the data flow. It ensures that both
+# query parameters (on the way in) and query results (on the way out) are
+# consistently strings, regardless of which `psycopg` implementation is in use.
+# This patch is applied globally and exactly once when this module is imported.
+# ==============================================================================
+from __future__ import annotations
+
+import json
+import logging
+from collections.abc import Iterator, Mapping, Sequence
+from contextlib import contextmanager
+from typing import TYPE_CHECKING, Any, cast
+
+import narwhals as nw
+import polars as pl
+from polars.datatypes import DataType as PolarsDataType
+from polars.datatypes import DataTypeClass as PolarsDataTypeClass
+from psycopg import Error as _PsycopgError
+from psycopg.cursor import Cursor as PsycopgCursor
+from typing_extensions import Self
+
+from metaxy.metadata_store.base import PROVENANCE_BY_FIELD_COL
+from metaxy.metadata_store.exceptions import HashAlgorithmNotSupportedError
+from metaxy.metadata_store.ibis import IbisMetadataStore
+from metaxy.metadata_store.types import AccessMode
+from metaxy.provenance.ibis import IbisProvenanceTracker
+from metaxy.provenance.types import HashAlgorithm, Increment, LazyIncrement
+from metaxy.utils.hashing import get_hash_truncation_length
+
+if TYPE_CHECKING:
+    from metaxy.metadata_store.base import MetadataStore
+    from metaxy.models.feature import BaseFeature
+    from metaxy.models.types import FeatureKey
+
+
+_PGCRYPTO_ERROR_TYPES: tuple[type[Exception], ...] = (_PsycopgError,)
+SchemaMapping = Mapping[str, PolarsDataType | PolarsDataTypeClass]
+
+logger = logging.getLogger(__name__)
+
+try:
+    if not getattr(PsycopgCursor, "_metaxy_compat_patched", False):
```

Comment:
I still need to dig into this issue, but I definitely don't feel good about just magically patching a global object to act differently. 

If this fix is truly needed, I can see 2 ways we could apply it: 
- scope it only to our metadata store. We should not affect other libraries and user code. 
- if that's not possible, at least make it opt-in via a configurable parameter somewhere